pkg/osquery/table/user_avatar_darwin.go

```diff
@@ -0,0 +1,138 @@
+package table
+
+/*
+#cgo darwin CFLAGS: -DDARWIN -x objective-c
+#cgo darwin LDFLAGS: -framework Cocoa -framework Collaboration
+#import <Collaboration/Collaboration.h>
+#include <CoreFoundation/CoreFoundation.h>
+void Image(CFDataRef *imageDataRef, char* user) {
+	NSString *userName = [NSString stringWithFormat:@"%s", user];
+	CBIdentity *identity = [CBIdentity identityWithName:userName authority:[CBIdentityAuthority defaultIdentityAuthority]];
+	NSImage *userImage = [identity image];
+	NSData *imageData = [userImage TIFFRepresentation];
+	*imageDataRef = (CFDataRef)imageData;
+}
+const char * LocalUsers() {
+	CSIdentityAuthorityRef defaultAuthority = CSGetDefaultIdentityAuthority();
+	CSIdentityQueryRef query = CSIdentityQueryCreate(NULL, kCSIdentityClassUser, defaultAuthority);
+	CFErrorRef error = NULL;
+	CSIdentityQueryExecute(query, 0, &error);
+	CFArrayRef results = CSIdentityQueryCopyResults(query);
+	int numResults = CFArrayGetCount(results);
+	NSMutableArray *usernames = [NSMutableArray array];
+	for (int i = 0; i < numResults; ++i) {
+ 		CSIdentityRef identity = (CSIdentityRef)CFArrayGetValueAtIndex(results, i);
+		CBIdentity *identityObject = [CBIdentity identityWithCSIdentity:identity];
+		[usernames addObject:[identityObject posixName]];
+	}
+	NSString *usernamesString = [usernames componentsJoinedByString:@" "];
+	CFRelease(results);
+	CFRelease(query);
+
+	const char *cString = [usernamesString UTF8String];
+	return cString;
+}
+*/
+import "C"
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"image"
+	"image/png"
+	"strings"
+	"unsafe"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/nfnt/resize"
+	"github.com/pkg/errors"
+	"golang.org/x/image/tiff"
+)
+
+func UserAvatar(logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("username"),
+		table.TextColumn("thumbnail"),
+	}
+	t := &userAvatarTable{logger: logger}
+	return table.NewPlugin("kolide_user_avatars", columns, t.generateAvatars)
+}
+
+type userAvatarTable struct {
+	logger log.Logger
+}
+
+func (t *userAvatarTable) generateAvatars(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	// use the username from the query context if provide, otherwise default to user created users
+	var usernames []string
+	q, ok := queryContext.Constraints["username"]
+	if ok && len(q.Constraints) != 0 {
+		for _, constraint := range q.Constraints {
+			usernames = append(usernames, constraint.Expression)
+		}
+	} else {
+		cString := C.LocalUsers()
+		usernamesString := C.GoString(cString)
```

Comment:
As a nit, I'd ditch `cSgtring` for `usernamesString := C.GoString(C.LocalUsers())`